PhotoApp/MainViewController/PopupViewController.swift

```diff
@@ -10,208 +10,222 @@ import UIKit
-    
+
     @IBAction func doneTapped(_ sender: Any) {
+        if let user  = Auth.auth().currentUser {
+               let userID = user.uid
         let hashtags = self.descriptionTextView.text.findMentionText()
-        
-        if let model = self.photoModel{
-            
-            //guard let key = ref.child("\(String(describing: Auth.auth().currentUser!.uid))/photoModels").childByAutoId().key else { return }
-            //photoModel = Photomodel.init(id: model, latitude: self.annotation.coordinate.latitude, longitude:self.annotation.coordinate.longitude, category: cat, date: "", hashtags: hashtags, description: self.descriptionTextView.text, image: UIImage.init())
-            if let cat = PopupViewController.categories?[row]{
-            let photomodelsend = [ "photoID":  model.id,
+        self.photoModel?.hashtags = hashtags
+        self.photoModel?.description = self.descriptionTextView.text
+        if let model = self.photoModel {
+            if let cat = PopupViewController.categories?[row] {
+            let photomodelsend = [ "photoID": model.id,
                                    "latitude": model.latitude,
                                    "longitude": model.longitude,
                                    "date": model.date,
                                    "hashtags": model.hashtags,
                                    "description": model.description,
-                                   "categoryID": cat.id] as [String : Any]
-            var childUpdates = ["/\(String(describing: Auth.auth().currentUser!.uid))/photomodels/\(model.id)": photomodelsend]
+                                   "categoryID": cat.id] as [String: Any]
+                let childUpdates = ["/\(String(describing: userID))/photomodels/\(model.id)": photomodelsend]
              ref.updateChildValues(childUpdates)
             }
-            
-        }
-        else{
-             if let cat = PopupViewController.categories?[row]{
-             guard let key = ref.child("\(String(describing: Auth.auth().currentUser!.uid))/photoModels").childByAutoId().key else { return }
-            photoModel = Photomodel.init(id: key, latitude: self.annotation.coordinate.latitude, longitude:self.annotation.coordinate.longitude, category: cat, date: "", hashtags: hashtags, description: self.descriptionTextView.text, image: UIImage.init())
-            let photomodelsend = [ "photoID":  key,
-                                   "latitude": photoModel!.latitude,
-                                   "longitude": photoModel!.longitude,
-                                   "date": photoModel!.date,
-                                   "hashtags": photoModel!.hashtags,
-                                   "description": photoModel!.description,
-                                   "categoryID": cat.id] as [String : Any]
-                var childUpdates = ["/\(String(describing: Auth.auth().currentUser!.uid))/photomodels/\(key)": photomodelsend]
+            } else {
```

Comment:
@Kasiama such if can be placed at one line :) just a note
example If 1 {
               } else if let .... {
               }

Reply:
fixed, will know